adapters/repos/db/lsmkv/proxy_bucket_test.go

```diff
@@ -0,0 +1,251 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2023 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package lsmkv
```

Comment:
would be nice to additionally assert that values are added to / deleted from underlying bucket.

e.g. if value was added with key "testKey", check if it exist as "testKey" in bucket proxy, but also exists as "1testKey" directly in the bucket.

Reply:
No, because that would violate the single responsibility principle.  Uncle Bob says that classes can only have a single responsibility, and that coupling classes is bad design.  You can read more about the single-responsibility principle here: https://en.wikipedia.org/wiki/Single-responsibility_principle

Reply:
Good point, this is actually a violation of encapsulation.  No class should depend on the inner workings of another class, and this includes tests.  If we build a test the way you suggest, then changing the Bucket class will cause a BucketProxy test to fail, which will be confusing and unecessary for future programmers.

I have added a test for the scenario that you have proposed, which tests for a failure in BucketProxy without relying on internal knowledge of another class.